src/main/java/org/gridsuite/study/server/service/RemoteServicesInspector.java

```diff
@@ -138,4 +164,28 @@ public CompletableFuture<JsonNode> getServiceInfo(@NonNull final RemoteServiceNa
             return CompletableFuture.completedFuture(NullNode.instance);
         }
     }
+
+    /**
+     * Aggregate result of {@link #getServicesInfo(FrontService)} into map of {@link AboutInfo}s for fronts
+     * @return a map for all services contacted
+     */
+    public AboutInfo[] convertServicesInfoToAboutInfo(@NonNull final Map<String, JsonNode> infos) {
+        return infos.entrySet().stream()
+            .map(e -> {
+                final JsonNode root = Objects.requireNonNullElse(e.getValue(), MissingNode.getInstance());
+                final String tag = JsonUtils.nodeAt(root, jn -> StringUtils.isNotBlank(jn.asText(null)),
+                        ACTUATOR_INFO_GIT_TAGS, ACTUATOR_INFO_GIT_COMMIT_DESCRIBE).asText(null);
+                return new AboutInfo(
+                        /*type*/ ModuleType.SERVER,
+                        /*name*/ JsonUtils.nodeAt(root,
```

Comment:
If you do it for the tag, do it here too ?
Actually extracting all the params into variables would be better IMO, then you can remove the comments

Reply:
that was just because Charly was complaining its not readable  
no intention from me to use variables here because I found them not necessary... unless you join him 😨

Reply:
Because you coded `nodeAt` and you know exactly what it does but not for someone who just arrive on this.
I agree it's easier to understand if you extract the params.

Reply:
okay...